cli/azd/docs/experimental-features.md

```diff
@@ -0,0 +1,54 @@
+# Experimental Features
+
+The Azure Developer CLI includes a `feature toggle` which allows the cli to ship `in-preview` ( _a.k.a. experimental_ ). This document provides the feature design for it.
+
+## Main requirement
+
+As a customer, I can explicitly ask `azd` to unblock and show any _experimental_ functionality. The features becoming available are not finalized, so they can be changed or removed in future releases.
+
+The strategy to ask `azd` to `toggle` between modes is:
+
+- Enable _all_ experimental features: `azd config set experimental on`
+- Enable _specific_ experimental feature: `azd config set experimental.name.feature`
+- Disable _all_ experimental features: `azd config unset experimental`
+- Disable _specific_ experimental feature: `azd config unset experimental.name.feature`
+
+## Experimental features
+
+The Azure Developer CLI offers the following type of features that can be toggled using experimental modes:
+
+1. New Command or sub-command
+    - The command becomes available to be run just like any other command.
+    - The command is designed exactly as how it would be released.
+      - There is no suffix or prefix to the command.
+      - There is no `experimental` command group which experimental commands live under.
+    - A `warning note` should be displayed then the command is run with a note like:
+      > You are running an experimental command. It can be changed or removed in the future.
+
+1. New Flags for command
+    - Follows the same rules from `1)`
+
+1. New Field for `azure.yaml`
+    - There won't be changes to the `yaml-schema`. This means, setting fields for in-preview features will be syntax highlighted as an error. Customer should expect and ignore those errors.
```

Comment:
It's probably worth noting that this is for Visual Studio Code, Codespaces, and VS.  We maybe even want to have a separate section for editor/IDE implications.

Reply:
Any json schema validator would complain about this. Not only IDE